tests/lib/c_lib/src/main.c

```diff
@@ -1236,3 +1236,14 @@ ZTEST(test_c_lib, test_exit)
 	k_thread_abort(tid);
 	zassert_equal(a, 0, "exit failed");
 }
+
+ZTEST(test_c_lib, test_clock)
+{
+	clock_t then;
+	clock_t now;
+
+	then = clock();
+	k_sleep(K_TICKS(1));
```

Comment:
I'd just use `k_msleep(1000)`; sleeping for a single tick relies on `CLOCKS_PER_SEC` being no less than `CONFIG_SYS_CLOCK_TICKS_PER_SEC`.